python/hail/linalg/blockmatrix.py

```diff
@@ -51,23 +51,94 @@ def read(cls, path):
         return cls(Env.hail().linalg.BlockMatrix.read(
             hc._jhc, path))
 
+    @classmethod
+    @typecheck_method(uri=str,
+                      n_rows=int,
+                      n_cols=int,
+                      block_size=nullable(int))
+    def fromfile(cls, uri, n_rows, n_cols, block_size=None):
+        """Creates a block matrix from a binary file.
+
+        Examples
+        --------
+
+        >>> import numpy as np
+        >>> a = np.random.rand(10, 20)
+        >>> a.tofile('/local/file') # doctest: +SKIP
+
+        To create a block matrix of the same dimensions:
+
+        >>> from hail.linalg import BlockMatrix
+        >>> bm = BlockMatrix.fromfile('file:///local/file', 10, 20) # doctest: +SKIP
+
+        Notes
+        -----
+        This method, analogous to `numpy.fromfile
+        <https://docs.scipy.org/doc/numpy/reference/generated/numpy.ndarray.fromfile.html>`__,
+        reads a binary file of float64 values in row-major order, such as that
+        produced by `numpy.tofile
+        <https://docs.scipy.org/doc/numpy/reference/generated/numpy.ndarray.fromfile.html>`__
+        or :meth:`BlockMatrix.tofile`.
```

Comment:
That link is broken.

What's more, from the fromfile documentation:

 > Do not rely on the combination of tofile and fromfile for data storage, as the binary files generated are are not platform independent. In particular, no byte-order or data-type information is saved.

You should make it the same point these are not portable and should only be used for interoperating with numpy, not storage.

You should also make it clear the matrix written on the numpy side had to have type float64 and that this is not checked (nor is it possible to check given the file format).